CONTRIBUTING.md

```diff
@@ -27,5 +31,48 @@ I recommend using [Visual Studio Code](https://code.visualstudio.com/) with
 
 
 
+## High-Level Documentation
+
+### Important Terms
+
+- CAS: Computer Algebra System, this is a piece of magic that takes formulas or equations and evaluates them
+- MathJson: [A format](https://cortexjs.io/math-json/) to unambiguously represent mathematical expressions
+- Document: A document consists of a number of elements such as
+  - Mathematical Expressions
+  - Scopes
+  - Text Blocks
+
+
+
+### Design
 
+- Equation Editor (frontend)
+  - We're using mathlive, however other equation editors could also be used
+  - Every equation editor has a corresponding element in the document
+  - During typing, the backend can optionally be asked to
+    - Return the variables (values, functions, units) that are in scope
+    - Quickly evaluate the expression
+    - Suggest autocompletions
+    - Return information about expressions, such as 'variable $a$ is a vector' or 'the $\cdot$ sign here stands for dot product'
+  - When the user hits Enter (submit), then
+    - The expression gets parsed and turned into MathJson
+      - Placeholders get added where the result should go
+    - The expression gets sent to the backend
+    - Eventually, the result gets returned
```

Comment:
Things that I'll have to check out again are:
- Should the backend only return the result or an entire new expression? 
- When do variables get resolved so that the frontend can highlight those? 
- Which part is responsible for turning $ab$ into $a \cdot b$ ?